clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -497,6 +551,32 @@ func (w *waitForControlledPodsRunningMeasurement) getObjectKeysAndMaxVersion() (
 	return objectKeys, maxResourceVersion, nil
 }
 
+type wrapperPodStore struct {
+	indexer       cache.Indexer
+	namespace     string
+	labelSelector labels.Selector
+}
+
+func (w *wrapperPodStore) List() []*v1.Pod {
+	objects, err := w.indexer.ByIndex("metadata.namespace", w.namespace)
+	if err != nil {
+		// This should only happen if index isn't registered, which effectively means programmers error.
+		klog.Errorf("podIndexer.ByIndex failed: %v", err)
+		return nil
+	}
+	pods := []*v1.Pod{}
+	for _, o := range objects {
+		pod := o.(*v1.Pod)
+		if w.labelSelector.Matches(labels.Set(pod.Labels)) {
```

Comment:
If this is our main performance bottleneck in this approach, maybe we can do something smarter here?

Obviously can hack this around by hardcoding the additional index for "name" (which would address load test), but we shouldn't do that. But maybe we can do something better, given we have "wrapperPodStore" per deployment?

If it would be keeping some cache (say from RV-> [yet/no] saying if a given object matches our selector or not), would that give us something?
It would mean that "Matches" would be called exactly once for each object version, but would acceessing the this cache (hash-map) actually be faster? (and won't cause significant memory increase)?

Or taking a step further:
- let's stay it would be storing: "objects that were matching selector in the previous call and RV from that point"
- then to get what is currently matching selector we can basically:
  (a) the the items that were previously matching selector and check if they are still in the list
  (b) check if objects with RV > "that previous RV" are matching selector

That makes the cache here proportional to the size of result (which means that memory impact should definitely be fine) and the comparisons should also be faster...

Reply:
As discussed offline, I decided to use ownerReference as index key. The presubmit has passed (it works in 100 node scale). I'm running 5k node test now to see performance gain.

It think this PR is ready for review.